### ISSUE TITLE
No armor for vampires

### ISSUE THREAD

**Issue by maxanier** [TeamLapen/Vampirism#287]
Vampires and armor do not really fit together. Vampires are fast and agile creatures whereas MC armors are always clunky and heavy.

Vampires already gain a strong boost in health and strength, so they should be able to survive without armor. Therefore I am currently considering somehow preventing vampires to wear armor parts. 
Instead they could be able to wear special rings or charms which give other boosts. However, I am expecting quite some resistance against this change, so I am not sure yet.

**Comment by GreatestFool** [comment]
Not sure if this has been implemented yet, but seeing as some bosses added by other mods essentially instantly kill the player unless they have the needed armor, I'd say this would be a bad idea.

However! I also like this idea, but if this is to be implemented, include a config option to disable this. If there is no way to disable the forced-no-armor-for-vampires then I'd say it shouldn't be implemented.

**Comment by maxanier** [comment]
Did not consider mods like that before.

If such feature is to be implemented there will definitively be a config option for that and maybe even some automatic disabling if certain mods are present.

**Comment by mopsyd** [comment]
Disable or throttle vampire abilities (helmets prevent night vision, chestplates prevent turning into a bat, etc) and possibly also add negative potion effects (mining fatigue, weakness, etc) instead of outright banning armor for them. That way armor can still be worn, however it will carry negative consequences that are not desirable except under specific circumstances where the armor is necessary.

The penalties should at least counteract any strength/defense buffs from vampirism for the duration that armor is worn, so that they never stack in a positive way with the armor.

**Comment by maxanier** [comment]
That is a way better idea. It makes a lot more sense and nullifies most of my concerns with the concept. Some people will still not like it, but that's ok.

_Note to myself regarding implementation:_
Guess we will have to figure out how to decide which armor parts can be worn without a negative effect, since decorative armor items (cloaks, hats, ...) should not have an impact. Probably just check the item's armor stats in addition to a configurable whitelist.
We probably also want to add a hint about the penalty to the corresponding items (only displayed to vampires of course)

**Comment by Octoham** [comment]
the vampire barons(lv 5) are hard even with enchanted diamond armor and enchanted diamond sword

**Comment by takashi1kun** [comment]
I have to mention, armor is not supposed to be clunky or slow, at least in a historic context, that is a common misconception in media. 
In battle the speed is always key, if you are slow your adversary can easily avoid your attacks and kill you no matter the armor. So armor is supposed to actually not be clunky nor slow you down at all. 
A trained knight is said to be able to move and maneuver at the same speed with or without armor, the main disadvantage from armor would be that it can be quite taxing to walk long distances wearing it, which is the main reason knights were usually paired with a horse because to march with armor can be quite taxing and they need to not be tired.
 
If we use logic, a vampire would actually be able to put even heavier armor than a human without losing speed since they have more strength. Plus they are kind of undead, so the main problem, ie being taxing, wouldn't be applicable because the whole undead are tireless. To end this, the other bad thing about armor, which is the danger of drowning if marching through water would also be obsolete since they do not breathe.

TL;DR Actually vampires would take even more advantage from armor than humans because they would be able to wear heavier armor(than humans) without losing their vampiric speed. So the main disadvantages of armor are kind of not applicable to vampires, if one they may become thirsty quicker while wearing armor to parallel how taxing it can be on humans. So it would be kind of not immersive for vampires to not be able to use armor, it would feel like a crap reason, since the no armor is not part of their folklore weaknesses and it would not make sense at all since vampires would actually profit from armor more than humans, so it would feel too much 'gamey' of a reason.

As an alternative take on this idea I propose this: 
You could prevent vampires from equipping enchanted armor. This would be more balanced than no armor for vampires, and the reason is much more simple and immersive: 
_"The magic from enchantments in armor surrounding your body interferes with your vampiric powers"_ 
To make it even funnier the restriction could manifest in the player blowing up when they try to use a vampire skill while wearing a piece of armor that is enchanted, kind of like sleeping in a bed in the nether/end.
This would even allow you to give an advantage to things like the vampire cloak or other possible future vampire armor, you could make the vampire armor weak kind like the cape in comparison with vanilla armors but with the advantage that it can be enchanted, you could say that it is a special kind of armor that sacrifices defense for the ability to channel both enchantments and vampiric power without interfering with each other, this would make so that vampiric players have to choose between equipping unenchanted netherite/diamond armor for the high defense or equipping enchanted vampire armor pieces, even if they themselves do not offer much protection but you could take advantage of enchantments.

P.D. Sorry for the bad English, i do not have the grammatically installed in this browser.

**Comment by maxanier** [comment]
While I agree that from a purely logical perspective it would make sense that vampires would benefit from armor in the same way as humans, but would suffer less disadvantages due to their increased base strength and speed.
And yes, in theory there can be flexible armor. But it will never be as flexible as no armor. Also armor in Vampirism always looks clunky due to the way it is rendered.

Anyway, for the mod the opposite would be more fun.
First of all from a balancing point of view, armor should be a way for hunters to face vampires in a fairer fight.
Secondly, I think it also makes sense that vampires would feel confined when wearing armor. Also things like teleporting or turning into a bat is hindered by that.

Therefore I propose the following concept: #908
